File: redome-front/src/app/shared/dto/exame.doador.internacional.dto.ts
import { LocusExame } from "../../paciente/cadastro/exame/locusexame";


export class ExameDoadorInternacionalDto{

    private _id: number;
    private _dataCriacao: Date;
    private _locusExames: LocusExame[] = [];
    private _statusExame:number;
    private _tipoExame:number;
    private _caminhoArquivo:string;
    private _idGenerico:number;
	constructor() {
	}



    /**
     * Getter idGenerico
     * @return {number}
     */
	public get idGenerico(): number {
		return this._idGenerico;
	}

    /**
     * Setter idGenerico
     * @param {number} value
     */
	public set idGenerico(value: number) {
		this._idGenerico = value;
	}

    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}


    /**
     * Getter dataCriacao
     * @return {Date}
     */
	public get dataCriacao(): Date {
		return this._dataCriacao;
	}

    /**
     * Setter dataCriacao
     * @param {Date} value
     */
	public set dataCriacao(value: Date) {
		this._dataCriacao = value;
	}

    /**
     * Getter locusExames
     * @return {LocusExame[] }
     */
	public get locusExames(): LocusExame[]  {
		return this._locusExames;
	}

    /**
     * Setter locusExames
     * @param {LocusExame[] } value
     */
	public set locusExames(value: LocusExame[] ) {
		this._locusExames = value;
	}


    /**
     * Getter statusExame
     * @return {number}
     */
	public get statusExame(): number {
		return this._statusExame;
	}

    /**
     * Setter statusExame
     * @param {number} value
     */
	public set statusExame(value: number) {
		this._statusExame = value;
    }


    /**
     * Getter tipoExame
     * @return {number}
     */
	public get tipoExame(): number {
		return this._tipoExame;
	}

    /**
     * Setter tipoExame
     * @param {number} value
     */
	public set tipoExame(value: number) {
		this._tipoExame = value;
	}


    /**
     * Getter caminhoArquivo
     * @return {string}
     */
	public get caminhoArquivo(): string {
		return this._caminhoArquivo;
	}

    /**
     * Setter caminhoArquivo
     * @param {string} value
     */
	public set caminhoArquivo(value: string) {
		this._caminhoArquivo = value;
    }

    public static get TIPO_EXAME_HLA(){
        return 1;
    }

    public static get TIPO_EXAME_IDM(){
        return 2;
    }

}

File: redome/src/main/java/br/org/cancer/modred/controller/dto/ExameDoadorInternacionalDto.java
package br.org.cancer.modred.controller.dto;

import java.time.LocalDateTime;
import java.util.List;

import br.org.cancer.modred.model.LocusExame;

/**
 * Dto para colocar exames de HLA e de IDM juntos.
 * 
 * @author Filipe Paes
 *
 */
public class ExameDoadorInternacionalDto {

	private Integer idGenerico;
	private Long id;
	private Integer statusExame = 0;
	private List<LocusExame> locusExames;
	private String caminhoArquivo;
	private Long tipoExame;
	private LocalDateTime dataCriacao;

	/**
	 * @return the idGenerico
	 */
	public Integer getIdGenerico() {
		return idGenerico;
	}

	/**
	 * @param idGenerico the idGenerico to set
	 */
	public void setIdGenerico(Integer idGenerico) {
		this.idGenerico = idGenerico;
	}

	/**
	 * @return the id
	 */
	public Long getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(Long id) {
		this.id = id;
	}

	/**
	 * @return the statusExame
	 */
	public Integer getStatusExame() {
		return statusExame;
	}

	/**
	 * @param statusExame
	 *            the statusExame to set
	 */
	public void setStatusExame(Integer statusExame) {
		this.statusExame = statusExame;
	}

	/**
	 * @return the locusExames
	 */
	public List<LocusExame> getLocusExames() {
		return locusExames;
	}

	/**
	 * @param locusExames
	 *            the locusExames to set
	 */
	public void setLocusExames(List<LocusExame> locusExames) {
		this.locusExames = locusExames;
	}

	/**
	 * @return the caminhoArquivo
	 */
	public String getCaminhoArquivo() {
		return caminhoArquivo;
	}

	/**
	 * @param caminhoArquivo
	 *            the caminhoArquivo to set
	 */
	public void setCaminhoArquivo(String caminhoArquivo) {
		this.caminhoArquivo = caminhoArquivo;
	}

	/**
	 * @return the tipoExame
	 */
	public Long getTipoExame() {
		return tipoExame;
	}

	/**
	 * @param tipoExame the tipoExame to set
	 */
	public void setTipoExame(Long tipoExame) {
		this.tipoExame = tipoExame;
	}

	/**
	 * @return the dataCriacao
	 */
	public LocalDateTime getDataCriacao() {
		return dataCriacao;
	}

	/**
	 * @param dataCriacao the dataCriacao to set
	 */
	public void setDataCriacao(LocalDateTime dataCriacao) {
		this.dataCriacao = dataCriacao;
	}

	
}
